app/controllers/cart_controller.rb

```diff
@@ -23,6 +23,46 @@ def remove_item
     session[:cart].delete(params[:item_id])
     redirect_to '/cart'
   end
+#this is the beginning of a refactor of the update method
+  # def update
+  #   require "pry"; binding.pry
+  #   case params[:add]
+  #   when true
+  #     if item.inventory_has_reached_limit?(cart)
+  #       flash[:error] = "You've reached the end of this merchant's inventory"
+  #       redirect_to '/cart'
+  #     else
+  #       cart.add_item(item.id.to_s)
+  #       redirect_to '/cart'
+  #     end
+  #   else
+  #
+  #   end
+  #
+  #   redirect_to "/cart"
+  #
+  #
+  # end
```

Comment:
Good job noticing the update action can be refactored - nested ifs are a big sign! Interested in seeing how this turns out.